src/App.tsx

```diff
@@ -1,23 +1,41 @@
 import { RouterProvider, createBrowserRouter } from "react-router-dom";
 import DashboardScreen from "./components/screens/dashboard/dashboard-screen";
 import ErrorPage from "./components/screens/error-page";
+import AuthenticationProvider from "./components/providers/authentication-provider";
+import { useState } from "react";
+import { type KeycloakProfile, type KeycloakTokenParsed } from "keycloak-js";
 
-const router = createBrowserRouter([
-  {
-    path: "/",
-    element: <DashboardScreen />,
-    errorElement: <ErrorPage />
-  }
-]);
+export interface Auth {
+  token: KeycloakTokenParsed | undefined
+  tokenRaw: string | undefined
+  logout: () => void
+}
 
 /**
  * Application component
  *
  */
 function App () {
+  const [auth, setAuth] = useState<Auth>();
```

Comment:
Even after seeing the discussions taken place in Slack, I'd implement this with Jotai state management since the beginning. Setup is so simple that I just simply don't see any drawbacks in doing that.

Reply:
This seems to be a task of its own in the Trello board. Will configure Jotai state management in a different issue and branch.